src/components/Edit.js

```diff
@@ -331,18 +331,24 @@ class Edit extends Component {
       })
 
     let timeoutID = null
-    doc.on('state changed', (fromSelf) => {
+    doc.shared.on('state changed', (fromSelf) => {
       if (fromSelf) {
         this.setState({stateStatus: stateStatuses.SAVING})
         clearTimeout(timeoutID)
         timeoutID = setTimeout(() => {
           this.setState({stateStatus: stateStatuses.TIMEOUT})
         }, SAVE_TIMEOUT_MS)
-      } else {
-        this.setState({stateStatus: stateStatuses.RECEIVING})
       }
     })
 
+    doc.replication.on('pinning', () => {
```

Comment:
I started out doing it like this (instead of using `state changed && !fromSelf` ), but the `pinning` event seems to be coming in too late, so it'll just be emitted right before `pinned` happens. Might be different if the state is very big.